nano/node/voting.cpp

```diff
@@ -79,14 +76,14 @@ void nano::vote_generator::run ()
 		}
 		else
 		{
-			wakeup = false;
-			if (!condition.wait_for (lock, node.config.vote_generator_delay, [this]() { return this->wakeup; }))
+			condition.wait_for (lock, node.config.vote_generator_delay, [this]() { return this->hashes.size () >= 12; });
+			if (hashes.size () >= node.config.vote_generator_threshold && hashes.size () < 12)
 			{
-				// Did not wake up early. Likely not under high load, ok to send lower number of hashes
-				if (!hashes.empty ())
-				{
-					send (lock);
-				}
+				condition.wait_for (lock, node.config.vote_generator_delay, [this]() { return this->hashes.size () >= 12; });
```

Comment:
Would it make sense to compute the additional wait as something like `(delay/12)*(12-hashes.size())` so it's inversely proportional to current hash count? Otherwise we'll wait an extra 100ms whether 1 or 9 hashes are missing for a full VBH. 

With default settings, for 3 hashes total wait would be 175ms ... 6 -> 150ms ... 11 -> ~110ms. IOW, extra wait is between 10 and 75 ms.

Reply:
That could be done, although `vote_generator::add` will wake it up from the sleep early if it reaches 12:
https://github.com/nanocurrency/nano-node/pull/2203/files#diff-6892810124346aa7a6e440e1e922d28eL26